tests/unit/components/playback/controls/AddCustomer.spec.js

```diff
@@ -120,9 +244,274 @@ describe("AddCustomer.enter_uuidbase57", () => {
+      const selector_id_suffix_nickname_id = "nickname-id";
+
+      const spied_text_validator_uuid = jest.spyOn(
+        TextValidation.prototype,
+        "validate_uuidBase_fiftyseven_encode"
+      );
+
+      const spied_text_validator_api = jest.spyOn(
+        TextValidation.prototype,
+        "validate_alphanumeric"
+      );
+
+      const spied_text_validator_nickname = jest.spyOn(
+        TextValidation.prototype,
+        "validate_nickname"
+      );
+
+      const spied_button_function = jest.spyOn(wrapper.vm, "clicked_button");
+
+      const target_input_field_uuid = wrapper.find(
+        "#input-widget-field-" + selector_id_suffix_alphanumeric_id
+      );
+      const target_error_message_uuid = wrapper.find(
+        "#input-widget-feedback-" + selector_id_suffix_alphanumeric_id
+      );
+      target_input_field_uuid.setValue(uuid);
+      await Vue.nextTick();
+      expect(spied_text_validator_uuid).toHaveBeenCalledWith(uuid);
+      expect(target_error_message_uuid.text()).toStrictEqual(invalid_uuid);
+
+      const target_input_field_apikey = wrapper.find(
+        "#input-widget-field-" + selector_id_suffix_apikey_id
+      );
+      const target_error_message_apikey = wrapper.find(
+        "#input-widget-feedback-" + selector_id_suffix_apikey_id
+      );
+      target_input_field_apikey.setValue(apikey);
+      await Vue.nextTick();
+      expect(spied_text_validator_api).toHaveBeenCalledWith(apikey);
+      expect(target_error_message_apikey.text()).toStrictEqual(invalid_apikey);
+
+      const target_input_field_nickname = wrapper.find(
+        "#input-widget-field-" + selector_id_suffix_nickname_id
+      );
+      const target_error_message_nickname = wrapper.find(
+        "#input-widget-feedback-" + selector_id_suffix_nickname_id
+      );
+      target_input_field_nickname.setValue(nickname);
+      await Vue.nextTick();
+      expect(spied_text_validator_nickname).toHaveBeenCalledWith(nickname);
+      expect(target_error_message_nickname.text()).toStrictEqual(
+        invalid_nickname
+      );
+
+      const target_button_label_btn = wrapper.findAll(".span__button_label");
+      const cancel_btn = target_button_label_btn.at(0);
+      expect(cancel_btn.attributes().style).toBe(
+        "color: rgb(255, 255, 255); width: 250px; left: 0px;"
+      );
+      const save_btn = target_button_label_btn.at(1);
+      expect(save_btn.attributes().style).toBe(save_btn_css);
+
+      await cancel_btn.trigger("click");
+      await Vue.nextTick();
+      expect(spied_button_function).toHaveBeenCalledWith(0);
+
+      await save_btn.trigger("click");
+      await Vue.nextTick();
+      expect(spied_button_function).toHaveBeenCalledWith(label);
+
+      const save_id_events = wrapper.emitted("save-id");
+      expect(save_id_events).toHaveLength(1);
+      expect(save_id_events[0]).toStrictEqual([
+        {
+          api_key: "06ad547f-fe02-477b-9473-f7977e4d5e17",
```

Comment:
have these values use the `test.each` variables you created at the top of the test

Reply:
Completed.